mono/utils/mono-state.h

```diff
@@ -20,7 +20,7 @@
 
 #define MONO_NATIVE_STATE_PROTOCOL_VERSION "0.0.1"
 
-MONO_BEGIN_DECLS
+G_BEGIN_DECLS
```

Comment:
G_BEGIN_DECLS isn't actually needed in this file at all. I'm leaving it in as we decided, but using `G` for private headers, `MONO` for public headers. It is a useful distinction, mostly  adhered to, though half optional.

Understood though on the individual functions vs. files *going forward*.
Eventually we can just remove this here.

Reply:
What I remember we decided is to not change what we don't need to. If that's what you are referring to by "I'm leaving it in as we decided", then you are already going against it. So either you don't touch it, or you fix it properly. I would strongly advise you to go for the former for sake of least effort 😉

Reply:
I view the decision as more of a risk/semantic/churn thing, because removing extern "C" is an actual semantic change, and they are strewn across many files.
Changing from `MONO` to `G` is a useful convention but w/o semantic change or risk, and smaller churn.
I'll get to it in a minute.
Longer term, the "correct" thing is probably removal, but that is a semantic/risk I'd rather put off.

Reply:
Reverted.